sound/soc/intel/common/soc-acpi-intel-tgl-match.c

```diff
@@ -394,6 +394,7 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_tgl_sdw_machines[] = {
 	},
 	{
 		.link_mask = 0x3, /* rt5682 on link0 & 2xmax98373 on link 1 */
+		.id = "CnviBt",
```

Comment:
what is this? an ACPI _HID? If yes, that's not a compliant HID and thus a NAK from me.

Reply:
Just use the ACPI_HID field to illustrate how the code works in the current patch - there is no real "CnviBt" programmed at coreboot/bios.

In other words, a new macro can be used to avoid using find_codec_info_acpi() for the BT offload dai link.

Reply:
Goodness. 
I have mixed feelings about this. On one hand, it's clearly an abuse of what this id was intended for, but on the other hand this hack is quite brilliant. I would never have thought about this...
Need to think about this, I am still not sold on the idea that we want to unconditionally add BT offload, we'd probably need something from platform firmware telling us to enable this link.

Reply:
I am not opposing the "hack" part, but what it must do is to **make sure** that user won't have an impression that this is a _HID and, if it suddenly appears in firmware or bootloader generated ACPI table, this can possible work. I.o.w. if you guarantee that
- user won't mistakenly recognize this as ACPI _HID
- this won't work in case the firmware / bootloader ACPI table has it

But I see a problem here. If by any chance somebody will abuse ACPI so much and actually put this as a name in the table, you will be in a position to have name collision.

Reply:
I also wonder if this is a platform-level information that needs to be stored in a table, or something that can be discovered based on CNVI PCI ID and/or flags/capabilities.

Reply:
So here are the 3 options:

1. Use this field with the fake non ACPI compliant name intentionally + add new quirk macro
2. Use quirk macro alone to resolve the DAI link as done in https://github.com/thesofproject/linux/pull/2774/commits/50b4529b2974d03ae75c1c2ab4451b4dde443ed0
3. Use genuine existing ACPI attribute if available, or add new one in cb for query.

Here is the BT-offload support implemented in coreboot:
https://review.coreboot.org/c/coreboot/+/45586/4/src/mainboard/google/volteer/variants/baseboard/devicetree.cb

Since machine driver is not board-blind, I feel option 1 and 2 are sufficient, but will follow the consensus reached here.